streams/src/main/java/org/apache/kafka/streams/state/internals/SessionStoreBuilder.java

```diff
@@ -50,7 +55,8 @@ public SessionStoreBuilder(final SessionBytesStoreSupplier storeSupplier,
         }
         return new CachingSessionStore(
             inner,
-            storeSupplier.segmentIntervalMs());
+            storeSupplier.segmentIntervalMs(),
+            dslUsage);
```

Comment:
I'm wondering if we have to pass in this boolean flag to `CachingSessionStore` or not; my understanding is that for caching session stores, whenever it has a flush listener and that listener is called, we should always use the window end timestamp, right?

Reply:
For the DSL yes, but not for the general case. PAPI uses should be allowed to define their own semantics IMHO.

Reply:
If a PAPI user 1) adds a store, and then 2) cast that store to a `CachedStateStore` and call `setFlushListener`, then when that listener is called, the timestamp is passed in and is not be controllable by the user still, right? Or how the users could specify which timestamp to pass in today?

Reply:
Thinking about this once more, I actually believe we don't need this flag and we can push all the logic into the flush-listener. Let me update this PR and we can discuss afterwards.